spec/api/keys/keys_spec.rb

```diff
@@ -0,0 +1,515 @@
+      original_data['public_key'] = public_key
+      put("#{platform.server}/users/#{user['name']}", superuser, :payload => JSON.generate(original_data))
+    end
+    context "when the default key has already been deleted" do
+      # TODO: things break
+    end
+    context "when the default key exists" do
+      it "should update the default key in the keys table" do
+        `chef-server-ctl list-user-keys #{user['name']}`.should include(public_key)
+      end
+      it "should no longer contain the old default key" do
+        `chef-server-ctl list-user-keys #{user['name']}`.should_not include(user['public_key'])
+      end
+      it "should return the new key from the /users endpoint" do
+        JSON.parse(get("#{platform.server}/users/#{user['name']}", superuser))['public_key'].should include(public_key)
+      end
+    end
+  end
+
+  context "when the default key is updated for a user via a PUT to /organizations/:org/clients" do
+    before do
+      original_data = JSON.parse(get("#{platform.server}/organizations/#{org['name']}/clients/#{client['name']}", superuser))
+      original_data['public_key'] = public_key
+      put("#{platform.server}/organizations/#{org['name']}/clients/#{client['name']}", superuser, :payload => JSON.generate(original_data))
+    end
+    context "when the default key has already been deleted" do
+      # TODO: things break
+    end
+    context "when the default key exists" do
+      it "should update the default key in the keys table" do
+        `chef-server-ctl list-client-keys #{org['name']} #{client['name']}`.should include(public_key)
+      end
+      it "should no longer contain the old default key" do
+        `chef-server-ctl list-client-keys #{org['name']} #{client['name']}`.should_not include(user['public_key'])
+      end
+      it "should return the new key from the /users endpoint" do
+        JSON.parse(get("#{platform.server}/organizations/#{org['name']}/clients/#{client['name']}", superuser))['public_key'].should include(public_key)
+      end
+    end
+  end
+
+  context "when a user and client with the same name exist" do
+    before do
+      # give user same name as client
+      delete("#{platform.server}/users/#{user['name']}", superuser)
+      user['name'] = client['name']
+      # post a user with the same name as the client, but with the other public key
+      payload = {
+        "username" => client['name'],
+        "first_name" => client['name'],
+        "middle_name" => client['name'],
+        "last_name" => client['name'],
+        "display_name" => client['name'],
+        "email" => "#{client['name']}@#{client['name']}.com",
+        "password" => "client-password",
+        "public_key" => user['public_key']
+      }
+      post("#{platform.server}/users", superuser, :payload => JSON.generate(payload))
+    end
+    after do
+      post("#{platform.server}/users/#{client['name']}", superuser)
+    end
+    it "should client" do
```

Comment:
I'd  also like to see something here for making sure the client can do something it's allowed to do but the user is not perhaps a PUT to the client itself? On the flip side, a 403 test for the user too, possibly the same PUT to the client.